src/OmniSharp.Roslyn.CSharp/Workers/Diagnostics/AnalyzerWorkQueue.cs

```diff
@@ -56,21 +56,21 @@ public void PutWork(IReadOnlyCollection<DocumentId> documentIds, AnalyzerWorkTyp
                 if (queue.WorkPendingToken == null)
                     queue.WorkPendingToken = new CancellationTokenSource();
 
-                queue.WorkWaitingToExecute = queue.WorkWaitingToExecute.Union(documentIds);
+                queue.WorkWaitingToExecute = queue.WorkWaitingToExecute.Union(documents);
```

Comment:
If an old version of the document exists in the queue, any idea what this will do?

1) Leave both documents (old and new)?
2) Leave the old document (union will check equality)?
3) Remove the old document and use the new document?

Logically I don't think 3 will ever happen, but I'm not sure about the equality behavior for 1 or 2.

Reply:
`Document`s are not equal across `Solution`s, so both would be in the queue. This is fine on its own, but I think it points out an issue that the caching layer shouldn't operate on the DocumentId's, but rather on the `Document`s themselves. The solution is to use a CWT, like Roslyn's analyzer driver, from `Document` to the diagnostics. I'll go ahead and do that.